### PR TITLE
Merge physical plan evaluator and basic planner to `main`

### PR DESCRIPTION
**Pull request by dlurton**
This will merge the following PRs to `main`:

- #583 
- #584 
- #587 
- #588 
- #589 
- #609 
- #590 
- #592 

Also, this adds a TODO comment (ea40e4a, requested by @am357 in offline chat) and excludes `PlannerPipeline` from the new aggregate tests which were pulled in when merging from `main` and were not part of the other PRs (34025b3).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by am357 — DISMISSED** [review]
LGTM, all the incorporating PRs have already been reviewed by maintainers.

Please wait for the second reviewer approval before merge. We want to have the 2-person verification for this merge.

File: lang/test/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransformTests.kt
Comment (alancai98) [inline lang/test/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransformTests.kt:121]:
```suggestion
     * A list of statements that cannot be converted into the logical algebra yet by [AstToLogicalVisitorTransform].  This
```

**Review by alancai98 — APPROVED** [review]
I gave a cursory look at all the changes. The logic based on the documentation and prior discussions looks reasonable to me, and I think it can be merged to `main`. The comments I left were all minor and mostly dealt with documentation.

File: lang/src/org/partiql/lang/SqlException.kt
Comment (dlurton) [reply lang/src/org/partiql/lang/SqlException.kt:39]: Removed

File: lang/src/org/partiql/lang/SqlException.kt
Comment (dlurton) [reply lang/src/org/partiql/lang/SqlException.kt:45]: It's nullable because it always has been nullable.  Changing this to an argument and defaulting `errorContext` to an empty map was a way of simplifying error handling code that reads context properties without changing all of the call sites (a few  direct, but many indirect) to conform it's new constructor signature, which would have resulted in many more noisy diffs in this series of PRs.  If you want, I can undo this change in another commit against this PR, but I would consider making this argument non-nullable to be out of scope of this work.

File: lang/test/org/partiql/lang/eval/EvaluatingCompilerOffsetTests.kt
Comment (dlurton) [reply lang/test/org/partiql/lang/eval/EvaluatingCompilerOffsetTests.kt:14]: removed.

File: lang/src/org/partiql/lang/eval/builtins/DynamicLookupExprFunction.kt
Comment (dlurton) [reply lang/src/org/partiql/lang/eval/builtins/DynamicLookupExprFunction.kt:31]: Done.

File: lang/src/org/partiql/lang/eval/physical/RelationThunk.kt
Comment (dlurton) [reply lang/src/org/partiql/lang/eval/physical/RelationThunk.kt:36]: This `EVALUATOR_GENERIC_EXCEPTION` is the code assigned by `DEFAULT_EXCEPTION_HANDLER_FOR_LEGACY_MODE` when the exception being rethrown is not an `EvaluationException` and therefore does not already have an error code.  

Since this is the same situation here, I don't see a reason why it would need a different error code.

File: lang/src/org/partiql/lang/eval/relation/RelationIterator.kt
Comment (dlurton) [reply lang/src/org/partiql/lang/eval/relation/RelationIterator.kt:20]: fixed.

File: lang/src/org/partiql/lang/planner/PlannerPipeline.kt
Comment (dlurton) [reply lang/src/org/partiql/lang/planner/PlannerPipeline.kt:99]: fxied.

File: lang/src/org/partiql/lang/eval/physical/RelationThunk.kt
Comment (dlurton) [reply lang/src/org/partiql/lang/eval/physical/RelationThunk.kt:19]: It is a little bit yucky that we `import`ing this just for `kdoc` references but... yeah... done.

File: lang/src/org/partiql/lang/planner/transforms/Util.kt
Comment (dlurton) [reply lang/src/org/partiql/lang/planner/transforms/Util.kt:10]: Fixing

File: lang/src/org/partiql/lang/planner/transforms/Util.kt
Comment (dlurton) [reply lang/src/org/partiql/lang/planner/transforms/Util.kt:18]: paritlq

File: lang/src/org/partiql/lang/planner/transforms/VariableIdAllocator.kt
Comment (dlurton) [reply lang/src/org/partiql/lang/planner/transforms/VariableIdAllocator.kt:15]: You mean mutable, but yes.  Thanks.

File: lang/test/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransformTests.kt
Comment (dlurton) [reply lang/test/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransformTests.kt:121]: applied.

File: lang/src/org/partiql/lang/eval/visitors/PartiqlPhysicalSanityValidator.kt
Comment (dlurton) [reply lang/src/org/partiql/lang/eval/visitors/PartiqlPhysicalSanityValidator.kt:23]: fixed

File: lang/src/org/partiql/lang/SqlException.kt
Comment (alancai98) [reply lang/src/org/partiql/lang/SqlException.kt:43]: Ok got it. I don't think it needs to be addressed in this PR. Could you make an issue and link here regarding making `errorContextArg` non-nullable?

File: lang/src/org/partiql/lang/planner/transforms/VariableIdAllocator.kt
Comment (alancai98) [reply lang/src/org/partiql/lang/planner/transforms/VariableIdAllocator.kt]: Yeah, I meant mutable. Thanks

**Review by alancai98 — APPROVED** [review]
Thanks for addressing the feedback. Given @am357's previous approval and the fact that my feedback was minor (mostly related to documentation), I think we can merge this feature branch into `main`.

**Review by alancai98 — APPROVED** [review]

**Pull request merged**
